WSRequestHandler.cpp

```diff
@@ -1093,4 +1095,286 @@ void WSRequestHandler::HandleGetRecordingFolder(WSRequestHandler* req)
 
 	req->SendOKResponse(response);
 	obs_data_release(response);
-}
\ No newline at end of file
+}
+
+void WSRequestHandler::HandleGetTextGDIPlusProperties(WSRequestHandler* req)
+{
+
+	const char* itemName = obs_data_get_string(req->data, "source");
+	if (!itemName)
+	{
+		req->SendErrorResponse("invalid request parameters");
+		return;
+	}
+
+	const char* sceneName = obs_data_get_string(req->data, "scene-name");
+	obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(sceneName);
+	if (scene == NULL) {
+		req->SendErrorResponse("requested scene doesn't exist");
+		return;
+	}
+
+	obs_sceneitem_t* sceneItem = Utils::GetSceneItemFromName(scene, itemName);
+	if (sceneItem != NULL)
+	{
+		obs_source_t* sceneItemSource = obs_sceneitem_get_source(sceneItem);
+		const char* sceneItemSourceId = obs_source_get_id(sceneItemSource);
+
+		if (strcmp(sceneItemSourceId, "text_gdiplus") == 0)
+		{
+			obs_data_t* response = obs_source_get_settings(sceneItemSource);
+			obs_data_set_string(response, "source", itemName);
+			obs_data_set_string(response, "scene-name", sceneName);
+			obs_data_set_bool(response, "render",
+				obs_sceneitem_visible(sceneItem));
+
+			obs_sceneitem_release(sceneItem);
+			req->SendOKResponse(response);
+			obs_data_release(response);
+		}
+		else
+		{
+			req->SendErrorResponse("not text gdi plus source");
+		}
+
+	}
+	else
+	{
+		req->SendErrorResponse("specified scene item doesn't exist");
+	}
+
+	obs_source_release(scene);
+
+}
+
+void WSRequestHandler::HandleSetTextGDIPlusProperties(WSRequestHandler* req)
+{
+	if (!req->hasField("source") )
+	{
+		req->SendErrorResponse("missing request parameters");
+		return;
+	}
+
+	const char* itemName = obs_data_get_string(req->data, "source");
+	if (!itemName)
+	{
+		req->SendErrorResponse("invalid request parameters");
+		return;
+	}
+
+	const char* sceneName = obs_data_get_string(req->data, "scene-name");
+	obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(sceneName);
+	if (scene == NULL) {
+		req->SendErrorResponse("requested scene doesn't exist");
+		return;
+	}
+
+	obs_sceneitem_t* sceneItem = Utils::GetSceneItemFromName(scene, itemName);
+	if (sceneItem != NULL)
+	{
+		obs_source_t* sceneItemSource = obs_sceneitem_get_source(sceneItem);
+		const char* sceneItemSourceId = obs_source_get_id(sceneItemSource);
+
+		if (strcmp(sceneItemSourceId, "text_gdiplus") == 0)
+		{
+
+			obs_data_t* settings = obs_source_get_settings(sceneItemSource);
+
+			if (req->hasField("align"))
+			{
+				obs_data_set_string(settings, "align", 
+					obs_data_get_string(req->data, "align"));
+			}
+
+			if (req->hasField("bk_color"))
+			{
+				obs_data_set_int(settings, "bk_color", 
+					(int)obs_data_get_int(req->data, "bk_color"));
```

Comment:
so I re-looked at obs-text.cpp and reviewed TextSource::Update only new_cat_lines and font_size get cast with int - I also noticed that bk_color and a few of the others should have used obs_data_get_uint32 - so I'll fix all those

obs-text.cpp line 638:

```
inline void TextSource::Update(obs_data_t *s)
{
	const char *new_text   = obs_data_get_string(s, S_TEXT);
	obs_data_t *font_obj   = obs_data_get_obj(s, S_FONT);
	const char *align_str  = obs_data_get_string(s, S_ALIGN);
	const char *valign_str = obs_data_get_string(s, S_VALIGN);
	uint32_t new_color     = obs_data_get_uint32(s, S_COLOR);
	uint32_t new_opacity   = obs_data_get_uint32(s, S_OPACITY);
	bool gradient          = obs_data_get_bool(s, S_GRADIENT);
	uint32_t new_color2    = obs_data_get_uint32(s, S_GRADIENT_COLOR);
	uint32_t new_opacity2  = obs_data_get_uint32(s, S_GRADIENT_OPACITY);
	float new_grad_dir     = (float)obs_data_get_double(s, S_GRADIENT_DIR);
	bool new_vertical      = obs_data_get_bool(s, S_VERTICAL);
	bool new_outline       = obs_data_get_bool(s, S_OUTLINE);
	uint32_t new_o_color   = obs_data_get_uint32(s, S_OUTLINE_COLOR);
	uint32_t new_o_opacity = obs_data_get_uint32(s, S_OUTLINE_OPACITY);
	uint32_t new_o_size    = obs_data_get_uint32(s, S_OUTLINE_SIZE);
	bool new_use_file      = obs_data_get_bool(s, S_USE_FILE);
	const char *new_file   = obs_data_get_string(s, S_FILE);
	bool new_chat_mode     = obs_data_get_bool(s, S_CHATLOG_MODE);
	int new_chat_lines     = (int)obs_data_get_int(s, S_CHATLOG_LINES);
	bool new_extents       = obs_data_get_bool(s, S_EXTENTS);
	bool new_extents_wrap  = obs_data_get_bool(s, S_EXTENTS_WRAP);
	uint32_t n_extents_cx  = obs_data_get_uint32(s, S_EXTENTS_CX);
	uint32_t n_extents_cy  = obs_data_get_uint32(s, S_EXTENTS_CY);

	const char *font_face  = obs_data_get_string(font_obj, "face");
	int font_size          = (int)obs_data_get_int(font_obj, "size");
	int64_t font_flags     = obs_data_get_int(font_obj, "flags");
	bool new_bold          = (font_flags & OBS_FONT_BOLD) != 0;
	bool new_italic        = (font_flags & OBS_FONT_ITALIC) != 0;
	bool new_underline     = (font_flags & OBS_FONT_UNDERLINE) != 0;
	bool new_strikeout     = (font_flags & OBS_FONT_STRIKEOUT) != 0;

	uint32_t new_bk_color   = obs_data_get_uint32(s, S_BKCOLOR);
	uint32_t new_bk_opacity = obs_data_get_uint32(s, S_BKOPACITY);
```